src/go/rdctl/cmd/shutdown.go

```diff
@@ -31,15 +48,144 @@ var shutdownCmd = &cobra.Command{
 		if err != nil {
 			return err
 		}
-		result, err := processRequestForUtility(doRequest("PUT", versionCommand("", "shutdown")))
+		cmd.SilenceUsage = true
+		result, err := doShutdown()
 		if err != nil {
 			return err
 		}
-		fmt.Println(string(result))
+		if result != nil {
+			fmt.Println(string(result))
+		}
 		return nil
 	},
 }
 
 func init() {
 	rootCmd.AddCommand(shutdownCmd)
+	shutdownCmd.Flags().BoolVar(&shutdownSettings.Verbose, "verbose", false, "be verbose")
+	shutdownCmd.Flags().BoolVar(&shutdownSettings.DontWait, "dont-wait", false, "don't wait for shutdown to be confirmed")
+}
+
+func doShutdown() ([]byte, error) {
```

Comment:
This looks complicated enough that we should probably move it out of `…/cmd`?

Reply:
Yup, good for another refactor...